perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java

```diff
@@ -256,9 +271,11 @@ public void setSendingEnabled(PerunSession sess, List<ApplicationMail> mails, bo
 		for (ApplicationMail mail : mails) {
 			// update sending (enabled / disabled)
 			if (Compatibility.isPostgreSql()) {
-				jdbc.update("update application_mails set send=? where id=?", (enabled) ? "1" : "0", mail.getId());
+				int existence = jdbc.update("update application_mails set send=? where id=?", (enabled) ? "1" : "0", mail.getId());
+				if (existence < 1) throw new ApplicationMailNotExistsException("Application mail does not exist.", mail);
```

Comment:
You should try/catch the ``jdbc.update()`` call. Handle ``EmptyResultDataAccessException`` as ``ApplicationMailNotExistsException`` and wrap ``RuntimeException`` to ``InternalErrorException``.